src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/TangibleAssetsNoteServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+import uk.gov.companieshouse.api.model.accounts.smallfull.SmallFullLinks;
+import uk.gov.companieshouse.api.model.accounts.smallfull.tangible.TangibleApi;
+import uk.gov.companieshouse.web.accounts.api.ApiClientService;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.tangible.TangibleAssets;
+import uk.gov.companieshouse.web.accounts.service.smallfull.TangibleAssetsNoteService;
+import uk.gov.companieshouse.web.accounts.transformer.smallfull.tangible.TangibleAssetsTransformer;
+import uk.gov.companieshouse.web.accounts.util.ValidationContext;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+@Service
+public class TangibleAssetsNoteServiceImpl implements TangibleAssetsNoteService {
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    @Autowired
+    private TangibleAssetsTransformer tangibleAssetsTransformer;
+
+    private static final UriTemplate SMALL_FULL_URI =
+        new UriTemplate(
+            "/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full");
+
+
+    private static final UriTemplate TANGIBLE_ASSET_NOTE =
+        new UriTemplate(
+            "/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/tangible-assets");
+
+    private static final String INVALID_URI_MESSAGE = "Invalid URI for tangible assets note resource";
+
+    @Override
+    public TangibleAssets getTangibleAssets(String transactionId, String companyAccountsId,
+        String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = TANGIBLE_ASSET_NOTE.expand(transactionId, companyAccountsId).toString();
+
+        TangibleApi tangibleApi;
+
+        try {
+            tangibleApi = apiClient.smallFull().tangible().get(uri).execute();
+        } catch (ApiErrorResponseException e) {
+
+            if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
+                return new TangibleAssets();
+            }
+
+            throw new ServiceException("Error retrieving tangible assets note resource", e);
+        } catch (URIValidationException e) {
+
+            throw new ServiceException(INVALID_URI_MESSAGE, e);
+        }
+
+        return tangibleAssetsTransformer.getTangibleAssets(tangibleApi);
+
+    }
+
+    @Override
+    public List<ValidationError> postTangibleAssets(String transactionId, String companyAccountsId,
+        TangibleAssets tangibleAssets, String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = TANGIBLE_ASSET_NOTE.expand(transactionId, companyAccountsId).toString();
+
+        String smallFullUri = SMALL_FULL_URI.expand(transactionId, companyAccountsId).toString();
```

Comment:
If you use the `getSmallFullAccounts` method in the `SmallFullService` this can go

Reply:
Resolved as it is inefficient to use the current `SmallFullService` method as it results in multiple ApiClient objects being created. This will likely be refactored in the future as it is not in the scope of the story.